### PR TITLE
S mat

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: src/HeckeTypes.jl [hunk 1]
```diff
@@ -312,69 +312,56 @@ end
 Type for rows of sparse matrices, to create one use
 `sparse_row`
 """
-mutable struct SRow{T}
+mutable struct SRow{T, S} # S <: AbstractVector{T}
```
Comment (fingolfin) [inline src/HeckeTypes.jl:315]: Any particular reason you are not making this explicit, i.e.:
```suggestion
mutable struct SRow{T, S <: AbstractVector{T}}
```

**Comment by thofma** [comment]
tests are failing

**Comment by fieker** [comment]
On Fri, Mar 17, 2023 at 07:00:56AM -0700, Tommy Hofmann wrote:
> Oscar tests are failing!
Intersting - clearly my fault, but looks ike an abuse: there should
never be 0 entries in a SMat, SRow

**Comment by thofma** [comment]
Still failing

**Pull request merged**